cpp/src/reductions/minmax.cu

```diff
@@ -131,125 +131,113 @@ struct create_minmax_with_nulls {
+    cudf::column_view const &col, rmm::mr::device_memory_resource *mr, cudaStream_t stream)
+  {
+    CUDF_FAIL("type not supported for minmax() op");
   }
 };
 
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<cudf::dictionary32>(const cudf::column_view &col,
-                               rmm::mr::device_memory_resource *mr,
-                               cudaStream_t stream)
-{
-  CUDF_FAIL("dictionary type not supported");
-}
-
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<cudf::string_view>(const cudf::column_view &col,
-                              rmm::mr::device_memory_resource *mr,
-                              cudaStream_t stream)
-{
-  CUDF_FAIL("string type not supported");
-}
-
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<cudf::list_view>(const cudf::column_view &col,
-                            rmm::mr::device_memory_resource *mr,
-                            cudaStream_t stream)
-{
-  CUDF_FAIL("list type not supported");
-}
-
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<cudf::struct_view>(const cudf::column_view &col,
-                              rmm::mr::device_memory_resource *mr,
-                              cudaStream_t stream)
-{
-  CUDF_FAIL("struct type not supported");
-}
-
-// unable to support fixed point due to DeviceMin/DeviceMax not supporting fixed point
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<numeric::decimal32>(const cudf::column_view &col,
-                               rmm::mr::device_memory_resource *mr,
-                               cudaStream_t stream)
-{
-  CUDF_FAIL("fixed-point type not supported");
-}
-
-template <>
-std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax_functor::
-operator()<numeric::decimal64>(const cudf::column_view &col,
-                               rmm::mr::device_memory_resource *mr,
-                               cudaStream_t stream)
-{
-  CUDF_FAIL("fixed-point type not supported");
-}
-
 }  // namespace
 
 std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> minmax(
-  const cudf::column_view &col,
+  cudf::column_view const &col,
   rmm::mr::device_memory_resource *mr = rmm::mr::get_current_device_resource(),
   cudaStream_t stream                 = 0)
 {
-  return type_dispatcher(col.type(), minmax_functor{}, col, mr, stream);
+  if (col.null_count() == col.size()) {
+    // this handles empty and all-null columns
+    // return scalars with valid==false
+    return {make_default_constructed_scalar(col.type()),
+            make_default_constructed_scalar(col.type())};
+  }
+  // if dictionary, we can run minmax on just the keys
+  auto const input_col = cudf::is_dictionary(col.type()) ? dictionary_column_view(col).keys() : col;
```

Comment:
This won't work for sliced dictionary columns right? The keys would be for the entire base column but the actual view of the column for which to get the minmax would be described by the indices and the offset applied, if any.

Reply:
Yes, you are right. Good catch.